src/Messages/OWSDisappearingMessagesJob.m

```diff
@@ -84,16 +88,20 @@ - (void)run
     uint64_t now = [NSDate ows_millisecondTimeStamp];
 
     __block uint expirationCount = 0;
-    [self.disappearingMessagesFinder enumerateExpiredMessagesWithBlock:^(TSMessage *message) {
-        // sanity check
-        if (message.expiresAt > now) {
-            DDLogError(@"%@ Refusing to remove message which doesn't expire until: %lld", self.tag, message.expiresAt);
-            return;
+    [self.storageManager.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
```

Comment:
I also moved `ensureDynamicInteractionsForThread` to the message view's `uiDatabaseConnection`.